daedalus/src/redemption_key_validation.js

```diff
@@ -0,0 +1,117 @@
+"use strict";
+
+// This functions mirrors purescript variants from daedalus/src/Daedalus/ClientApi.purs
+// This is a part of rewrite from purescript into js and depricating purescript bits
+
+const bs58 = require("bs58");
+
+// Decode a Base64-encoded string using Node's `Buffer` API. Its trying to mimic atob functionality available in browsers.
+// This runs default node facility to decode base64 but additionally runs base64 validation code.
+// Note that atob under the hood runs validation. See example:
+//  > atob("====")
+//  InvalidCharacterError: String contains an invalid character
+//  > atob("aa==")
+//  "i"
+//  > atob("aa=")
+//  InvalidCharacterError: String contains an invalid character
+//
+//  Where node's implementation doesn't:
+//  > Buffer.from("====", "base64").toString("utf-8")
+//  ''
+//  > Buffer.from("aa==", "base64").toString("utf-8")
+//  'i'
+//  > Buffer.from("aa=", "base64").toString("utf-8")
+//  'i'
+//
+// And that's the reason this funciton exists: to extend Node's base64 decode with validity check.
+function decodeNode (str) {
```

Comment:
This probably won't be run in Electron as its a browser box with `atob` defined.